File: day4/compiled-codes/04-f2py/ex02/sum.py
#!/usr/bin/env python

from sum import *

print("Calling fortran with ints")
isum = sum_of_int(1,2)
print("Integer sum is: ", isum)

dsum = sum_of_double(1,2)
print("Double sum of ints is: ", dsum)

dsum = sum_of_double(0.5,2.5)
print("Double sum of doubles is: ", dsum)

print("Done")

File: day4/compiled-codes/04-f2py/ex02/sum.f90

function sum_of_int(a,b) result(c)
    integer, intent(in) :: a, b
    integer :: c

    c = a + b
    print*,"sum of ", a, " and ", b, " is ", c
end function sum_of_int

function sum_of_double(a,b) result(c)
    double precision, intent(in) :: a, b
    double precision :: c

    c = a + b
    print*,"sum of ", a, " and ", b, " is ", c
end function sum_of_double
